includes/class-cron-options-cpt.php

```diff
@@ -0,0 +1,375 @@
+		}
+
+		return $old_value;
+	}
+
+	/**
+	 * Delete jobs that are unscheduled using `wp_unschedule_event()`
+	 */
+	private function unschedule_job( $new_value, $old_value ) {
+		$jobs = $this->find_unscheduled_jobs( $new_value, $old_value );
+
+		foreach ( $jobs as $job ) {
+			$this->delete_job( $job['timestamp'], $job['action'], $job['instance'] );
+		}
+	}
+
+	/**
+	 * Save cron events in CPT
+	 */
+	private function convert_option( $new_value ) {
+		if ( is_array( $new_value ) && ! empty( $new_value ) ) {
+			$events = collapse_events_array( $new_value );
+
+			foreach ( $events as $event ) {
+				$job_exists = $this->job_exists( array(
+					'name'             => $this->event_name( $event['timestamp'], $event['action'], $event['instance'] ),
+					'post_type'        => self::POST_TYPE,
+					'post_status'      => self::POST_STATUS,
+					'suppress_filters' => false,
+					'posts_per_page'   => 1,
+				) );
+
+				if ( ! $job_exists ) {
+					// Build minimum information needed to create a post
+					$job_post = array(
+						'post_title'            => $this->event_title( $event['timestamp'], $event['action'], $event['instance'] ),
+						'post_name'             => $this->event_name( $event['timestamp'], $event['action'], $event['instance'] ),
+						'post_content_filtered' => maybe_serialize( array(
+							'action'   => $event['action'],
+							'instance' => $event['instance'],
+							'args'     => $event['args'],
+						) ),
+						'post_date'             => date( 'Y-m-d H:i:s', $event['timestamp'] ),
+						'post_date_gmt'         => date( 'Y-m-d H:i:s', $event['timestamp'] ),
+						'post_type'             => self::POST_TYPE,
+						'post_status'           => self::POST_STATUS,
+					);
+
+					$this->create_job( $job_post );
+				}
+			}
+		}
+	}
+
+	/**
+	 * PLUGIN UTILITY METHODS
+	 */
+
+	/**
+	 * Retrieve list of jobs, respecting whether or not the CPT is registered
+	 */
+	private function get_jobs( $args ) {
+		// If called before `init`, we need to query directly because post types aren't registered earlier
+		if ( did_action( 'init' ) ) {
+			return get_posts( $args );
+		} else {
+			global $wpdb;
+
+			$orderby = 'date' === $args['orderby'] ? 'post_date' : $args['orderby'];
+
+			if ( isset( $args['paged'] ) ) {
+				$paged  = max( 0, $args['paged'] - 1 );
+				$offset = $paged * $args['posts_per_page'];
+			} else {
+				$offset = 0;
+			}
+
+			return $wpdb->get_results( $wpdb->prepare( "SELECT * FROM {$wpdb->posts} WHERE post_type = %s AND post_status = %s ORDER BY %s %s LIMIT %d,%d;", $args['post_type'], $args['post_status'], $orderby, $args['order'], $offset, $args['posts_per_page'] ), 'OBJECT' );
+		}
+	}
+
+	/**
+	 * Check if a job post exists, respecting Core's loading order
+	 */
+	private function job_exists( $job_post ) {
+		// If called before `init`, we need to insert directly because post types aren't registered earlier
+		if ( did_action( 'init' ) ) {
+			$exists = get_posts( $job_post );
```

Comment:
Good call, `get_page_by_path()` is safe to use before init, and there's some added caching too.